### PR TITLE
templates: Use domain for accessing the etcd endpoint

### PR DESCRIPTION
**Pull request by nhlfr**

### CODE REVIEW

File: templates.go [hunk 1]
```diff
@@ -524,13 +524,13 @@ coreos:
       ExecStartPre=/bin/bash -c "while [ ! -f /etc/kubernetes/ssl/etcd/server-ca.pem ]; do echo 'Waiting for /etc/kubernetes/ssl/etcd/server-ca.pem to be written' && sleep 1; done"
       ExecStartPre=/bin/bash -c "while [ ! -f /etc/kubernetes/ssl/etcd/server-crt.pem ]; do echo 'Waiting for /etc/kubernetes/ssl/etcd/server-crt.pem to be written' && sleep 1; done"
       ExecStartPre=/bin/bash -c "while [ ! -f /etc/kubernetes/ssl/etcd/server-key.pem ]; do echo 'Waiting for /etc/kubernetes/ssl/etcd/server-key.pem to be written' && sleep 1; done"
-      ExecStart=/usr/bin/etcd2 --advertise-client-urls=https://127.0.0.1:2379,https://127.0.0.1:2380 \
+      ExecStart=/usr/bin/etcd2 --advertise-client-urls=https://0.0.0.0:2379,https://0.0.0.0:2380 \
                                --data-dir=/etc/kubernetes/data/etcd/ \
-                               --initial-advertise-peer-urls=https://127.0.0.1:2380 \
-                               --listen-client-urls=https://0.0.0.0:2379,https://127.0.0.1:2380 \
+                               --initial-advertise-peer-urls=https://0.0.0.0:2380 \
+                               --listen-client-urls=https://0.0.0.0:2379,https://0.0.0.0:2380 \
```
Comment (JosephSalisbury) [inline templates.go]: For my learning, why do we use both `2379` and `2380` here?

File: templates.go [hunk 2]
```diff
@@ -598,15 +598,15 @@ coreos:
...
+      ExecStartPre=/bin/bash -c "while ! curl --output /dev/null --silent --fail --cacert /etc/kubernetes/ssl/calico/client-ca.pem --cert /etc/kubernetes/ssl/calico/client-crt.pem --key /etc/kubernetes/ssl/calico/client-key.pem https://{{ .Cluster.Etcd.Domain }}:2379/version; do sleep 1 && echo 'Waiting for etcd master to be responsive'; done"
       ExecStartPre=/opt/bin/calicoctl pool add {{.Cluster.Calico.Subnet}}/{{.Cluster.Calico.CIDR}} --ipip --nat-outgoing
       ExecStart=/opt/bin/calicoctl node --ip=${DEFAULT_IPV4}  --detach=false --node-image=giantswarm/node:v0.22.0
       ExecStartPost=/bin/bash -c "/opt/bin/calicoctl bgp peer add $(echo ${BRIDGE_IP} | cut -d'.' -f1-3).0 as $(/opt/bin/calicoctl bgp default-node-as)"
-      ExecStartPost=/bin/bash -c "/usr/bin/etcdctl --endpoints=http://127.0.0.1:2380 set /calico/v1/host/{{.Node.Hostname}}-flannel/bird_ip $(echo ${BRIDGE_IP} | cut -d'.' -f1-3).0"
+      ExecStartPost=/bin/bash -c "/usr/bin/etcdctl --endpoints=http://{{ .Cluster.Etcd.Domain }}:2380 set /calico/v1/host/{{.Node.Hostname}}-flannel/bird_ip $(echo ${BRIDGE_IP} | cut -d'.' -f1-3).0"
```
Comment (JosephSalisbury) [inline templates.go]: This should be `2379`, no?

File: templates.go [hunk 3]
```diff
@@ -598,15 +598,15 @@ coreos:
...
       ExecStop=/opt/bin/calicoctl node stop --force
       ExecStopPost=/bin/bash -c "find /tmp/ -name '_MEI*' | xargs -I {} rm -rf {}"
       ExecStopPost=/bin/bash -c "/opt/bin/calicoctl bgp peer remove $(echo ${BRIDGE_IP} | cut -d'.' -f1-3).0"
-      ExecStopPost=/usr/bin/etcdctl --endpoints=http://127.0.0.1:2380 rm /calico/v1/host/{{.Node.Hostname}}-flannel/bird_ip
+      ExecStopPost=/usr/bin/etcdctl --endpoints=http://{{ .Cluster.Etcd.Domain }}:2380 rm /calico/v1/host/{{.Node.Hostname}}-flannel/bird_ip
```
Comment (JosephSalisbury) [inline templates.go]: This should be `2379`, no?

File: templates.go (hunk 1 above)
Comment (JosephSalisbury) [inline templates.go]: My brain feels like it should be just `2379`

Comment (asymmetric) [inline templates.go]: I think we just took it from [here](https://github.com/giantswarm/k8scloudconfig/blob/8a20e5e3895796d91b4f1b7dad9d68e0cab7db04/templates.go#L527), without really knowing the reason behind it.

Comment (nhlfr) [inline templates.go]: done, I'm using just `2379`

File: templates.go (hunk 2 above)
Comment (nhlfr) [inline templates.go]: done

File: templates.go (hunk 3 above)
Comment (nhlfr) [inline templates.go]: done

**Pull request merged**